components/server/src/api/auth-provider-service-api.ts

```diff
@@ -47,20 +50,20 @@ export class AuthProviderServiceAPI implements ServiceImpl<typeof AuthProviderSe
         }
 
         if (organizationId) {
-            const result = await this.authProviderService.createOrgAuthProvider(context.user.id, {
+            const result = await this.authProviderService.createOrgAuthProvider(ctxUserId(), {
                 organizationId,
                 host: request.host,
-                ownerId: context.user.id,
+                ownerId: ctxUserId(),
                 type: this.apiConverter.fromAuthProviderType(request.type),
                 clientId: request.oauth2Config?.clientId,
                 clientSecret: request.oauth2Config?.clientSecret,
             });
 
             return new CreateAuthProviderResponse({ authProvider: this.apiConverter.toAuthProvider(result) });
         } else {
-            const result = await this.authProviderService.createAuthProviderOfUser(context.user.id, {
+            const result = await this.authProviderService.createAuthProviderOfUser(ctxUserId(), {
                 host: request.host,
-                ownerId: context.user.id,
```

Comment:
indeed, this can be `ownerId`.

Reply:
actually, let me think. I believe this needs to be the `context.user.id`/`ctxUserId` but validated to be same as `ownerId`.

here were talking about (obsolete) user-level providers, which can only be created for the current user.

Reply:
Yeeeeah, was thinking in the same direction.

Reply:
is `ctxUserId` still derived from `sub`(ject) of the JWT cookie? if so, this diff look good to me. at least, I'm not keen to change to `ownerId`.

```
oneOf owner {
  ownerId string
  orgId string
}
```
feels to be not 100% correctly defined for `createAuthProvider` params. see, there is only a way to create this type of resource with owner being the current user or the specified org.

Reply:
We can reconsider and remove oneof, just organization_id if not provided then a user?

Reply:
We synced on this, and decided to phase out the ownerId. This PR removes the validation/handling of it, @AlexTugarev will follow-up with a change to the proto file.